### PR TITLE
Feature/reactify data portal

### PR DESCRIPTION
**Pull request by tiagojsag**
Fixes https://github.com/Vizzuality/trase/issues/126

Please test this carefully @sorodrigo

### CODE REVIEW

**Comment by sorodrigo** [comment]
I rebased against #208 so let's merge this after that. Not just yet, I still have to review it.

File: frontend/scripts/pages/data.page.jsx [hunk 1]
```diff
@@ -2,24 +2,31 @@
...
-  store.dispatch(loadContext());
 
+  render(
+    <Provider store={store}>
+      <DataPortalPage />
+    </Provider>,
+    document.querySelector('.c-container')
+  );
+
+  store.dispatch(loadContext());
```
Comment (sorodrigo) [inline frontend/scripts/pages/data.page.jsx]: I moved this to a `.thunks.js` file. Let's strive to maintain the `.page.js` files as dumb as possible.

File: frontend/scripts/react-components/data/bulk-downloads-block.component.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        <div className="c-bulk-downloads__title">BULK DOWNLOADS</div>
+        <div className="row">
+          {this.props.contexts.map((context, key) => (
+            <div key={key} className="small-4 columns">
+              <div
+                className={classnames(
+                  'c-bulk-downloads__item',
+                  { '-disabled': !this.props.enabled }
+                )}
+                data-value="{context.id}"
```
Comment (sorodrigo) [inline frontend/scripts/react-components/data/bulk-downloads-block.component.jsx]: is this used somewhere?

File: frontend/scripts/react-components/data/data-portal-page.component.jsx [hunk 3]
```diff
@@ -0,0 +1,531 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React, { Component } from 'react';
+import _ from 'lodash';
```
Comment (sorodrigo) [inline frontend/scripts/react-components/data/data-portal-page.component.jsx]: use individual imports

File: frontend/scripts/react-components/data/data-portal-page.component.jsx [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+
+      outputType: 'pivot',
+      fileExtension: '.csv',
+      fileSeparator: 'comma',
+
+      currentDownloadParams: null,
+      currentDownloadType: null,
+
+      downloaded: false
+    };
```
Comment (sorodrigo) [inline frontend/scripts/react-components/data/data-portal-page.component.jsx]: This component feels super bloated, the methods are doing too much for a component. I'd consider extracting the logic to the container and the store where possible. I know that there's a lot of different (and edge) cases we need to cover but it's pretty hard to follow the code.

File: frontend/scripts/react-components/data/data-portal-page.component.jsx [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+      id: indicator.name,
+      name: `${indicator.frontendName}${indicator.unit !== null ? `(${indicator.unit})` : ''}`,
+      noSelfCancel: false
+    }))
+      .sort((a, b) => a.name.localeCompare(b.name));
+
+    return (
+      <div className="l-data">
+        {DATA_DOWNLOAD_ENABLED === false &&
+        <DataPortalDisabledMessage />
```
Comment (sorodrigo) [inline frontend/scripts/react-components/data/data-portal-page.component.jsx]: this component seems like an overkill

File: frontend/scripts/react-components/data/download-selector.component.jsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  render() {
+    return (
+      <div
+        className={classnames(
+          'c-custom-dataset-selector',
+          { '-disabled': !this.props.enabled }
+        )}
+        data-type="value"
```
Comment (sorodrigo) [inline frontend/scripts/react-components/data/download-selector.component.jsx]: do we need this?

File: frontend/scripts/utils/getURLFromParams.js [hunk 7]
```diff
@@ -70,6 +70,7 @@ function getURLForV3(endpoint, paramsArg = {}) {
   const queryParams = Object.keys(params).reduce((prev, current) => {
```
Comment (sorodrigo) [inline frontend/scripts/utils/getURLFromParams.js]: why no use `query-string`

Comment (tiagojsag) [reply frontend/scripts/utils/getURLFromParams.js]: sounds like a good idea, but maybe for another day

**Comment by sorodrigo** [comment]
Thanks! I'll add a ticket about the component refactor for future tiago and future rodrigo.
651a07916ea9eaa9aaf78c4c6afe02a3
